src/coreclr/jit/hwintrinsicxarch.cpp

```diff
@@ -3379,7 +3379,17 @@ GenTree* Compiler::impSpecialIntrinsic(NamedIntrinsic        intrinsic,
 
             if (varTypeIsByte(simdBaseType))
             {
-                // byte and sbyte would require more work to support
+                op2 = impPopStack().val;
+                op1 = impSIMDPopStack();
+
+                GenTree* andShiftVal = gtNewOperNode(GT_AND, op2->TypeGet(), op2, gtNewIconNode(0x7));
+                op3                  = gtNewSimdBinOpNode(GT_LSH, retType, op1, andShiftVal,
+                                         varTypeIsUnsigned(simdBaseType) ? CORINFO_TYPE_USHORT : CORINFO_TYPE_SHORT,
+                                                          simdSize);
+                GenTree* maskElement =
+                    gtNewOperNode(GT_LSH, andShiftVal->TypeGet(), gtNewIconNode(0xFF), gtCloneExpr(andShiftVal));
+                GenTree* mask        = gtNewSimdCreateBroadcastNode(retType, maskElement, simdBaseJitType, simdSize);
+                retNode              = gtNewSimdBinOpNode(GT_AND, retType, op3, mask, simdBaseJitType, simdSize);
```

Comment:
We should rather have this handled in `gtNewSimdBinOpNode` so that any code can reuse the logic.

The branch here is meant to be an early out until that handling could be added.

Reply:
I see, that makes sense. I can move this logic to `gtNewSimdBinOpNode` instead.